storage.go

```diff
@@ -3,34 +3,187 @@ package ftp
 import (
 	"context"
 	"io"
+	"path/filepath"
 
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
 	. "github.com/beyondstorage/go-storage/v4/types"
+	"github.com/jlaffaye/ftp"
+	mime "github.com/qingstor/go-mime"
 )
 
+type listDirInput struct {
+	rp  string
+	dir string
+
+	started           bool
+	continuationToken string
+	objChan           []*ftp.Entry
+	counter           int
+	buf               []byte
+}
+
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(false)
+		o.Mode = ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode = ModeRead
+	}
+
+	o.ID = filepath.Join(s.workDir, path)
+	o.Path = path
+	return o
+}
+func (s *Storage) createDir(ctx context.Context, path string) (o *Object, err error) {
```

Comment:
`createDir` is the method implemented for `Direr`. Since it can be added later, it's OK for me to just keep this method here.